### PR TITLE
feat: Backfill Transaction feature added

### PR DESCRIPTION
**Pull request by vishdha**
BackFill Transactions:
Now that we’re syncing transactions, it’s likely that you have previous orders that haven’t been synced to TaxJar yet. To backfill transactions, click the Push Transactions button.

Note: Create Transaction should be checked in TaxJar Configuration.

Transaction-Import-on-2021-12-06-03-47-09-959927

Force: When Force checkbox is checked then it syncs all the taxjar transactions from date range. It also fetches all sales invoice transactions from Bloomstack.
It updates matching transactions and creates non-existing transactions.

If the Force checkbox is unchecked then it creates non-existing transactions to taxjar.

After selecting the From and To states, click the green Push Transaction button. We’ll look for orders sync within that date range and push them to TaxJar. Once the process is finished, you’ll be able to view a log of what happened.
Whenever a transaction is synced to TaxJar, it’s recorded in a log file for your reference. In Integration Request.
Peek 2021-12-06 21-57

### CODE REVIEW

File: erpnext/erpnext_integrations/taxjar_integration.py [hunk 1]
```diff
@@ -111,6 +114,59 @@ def delete_transaction(doc, method):
...
+			response = client.update_refund(doc.name, tax_dict)
+		else:
+			update_integration_request(integration_request, endpoint="client.update_order")
+			response = client.update_order(doc.name, tax_dict)
+	except taxjar.exceptions.TaxJarResponseError as err:
+		update_integration_request(integration_request, status="Failed", error=sanitize_error_response(err))
+		error_msg = """There seems to be an error in Address<br><br>{0}""".format(sanitize_error_response(err))
+		frappe.throw(_(error_msg), AddressError, _("Address Error"))
+	# except Exception as ex:
+	# 	print(traceback.format_exc(ex))
```
Comment (Anurag810) [inline erpnext/erpnext_integrations/taxjar_integration.py]: `<br><br>`  will cause problem during translation. will recommend adding tags on translated string.

File: erpnext/erpnext_integrations/taxjar_integration.py [hunk 2]
```diff
@@ -111,6 +114,59 @@ def delete_transaction(doc, method):
...
+			tax_dict['transaction_reference_id'] = doc.return_against
+			response = client.update_refund(doc.name, tax_dict)
+		else:
+			update_integration_request(integration_request, endpoint="client.update_order")
+			response = client.update_order(doc.name, tax_dict)
+	except taxjar.exceptions.TaxJarResponseError as err:
+		update_integration_request(integration_request, status="Failed", error=sanitize_error_response(err))
+		error_msg = """There seems to be an error in Address<br><br>{0}""".format(sanitize_error_response(err))
+		frappe.throw(_(error_msg), AddressError, _("Address Error"))
+	# except Exception as ex:
```
Comment (Anurag810) [inline erpnext/erpnext_integrations/taxjar_integration.py]: remove commented code

File: erpnext/erpnext_integrations/taxjar_integration.py [hunk 3]
```diff
@@ -77,11 +80,11 @@ def create_transaction(doc, method):
...
 			response = client.create_order(tax_dict)
 	except taxjar.exceptions.TaxJarResponseError as err:
-		update_integration_request(integration_request, status="Failed", error=str(err))
-		error_msg = """There seems to be an error in Address<br><br>{0}""".format(sanitize_error_response(err))
-		frappe.throw(_(error_msg), AddressError, _("Address Error"))
-	except Exception as ex:
-		print(traceback.format_exc(ex))
+		update_integration_request(integration_request, status="Failed", error=sanitize_error_response(err))
+		error_msg = """There seems to be an error <br><br>{0}""".format(sanitize_error_response(err))
+		# frappe.throw(_(error_msg))
```
Comment (Anurag810) [inline erpnext/erpnext_integrations/taxjar_integration.py]: remove commented code.

File: erpnext/erpnext_integrations/taxjar_integration.py [hunk 4]
```diff
@@ -77,11 +80,11 @@ def create_transaction(doc, method):
 			update_integration_request(integration_request, endpoint="client.create_order")
 			response = client.create_order(tax_dict)
 	except taxjar.exceptions.TaxJarResponseError as err:
-		update_integration_request(integration_request, status="Failed", error=str(err))
-		error_msg = """There seems to be an error in Address<br><br>{0}""".format(sanitize_error_response(err))
-		frappe.throw(_(error_msg), AddressError, _("Address Error"))
-	except Exception as ex:
-		print(traceback.format_exc(ex))
+		update_integration_request(integration_request, status="Failed", error=sanitize_error_response(err))
+		error_msg = """There seems to be an error <br><br>{0}""".format(sanitize_error_response(err))
```
Comment (Anurag810) [inline erpnext/erpnext_integrations/taxjar_integration.py]: use _() to translate the string and `<br><br>`  will cause problems during translation. will recommend to add tags on a translated string.

File: erpnext/erpnext_integrations/doctype/backfill_taxjar_transactions/backfill_taxjar_transactions.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ "creation": "2021-11-10 03:20:52.588269",
+ "doctype": "DocType",
+ "editable_grid": 1,
+ "engine": "InnoDB",
+ "field_order": [
+  "company",
+  "column_break_2",
+  "status",
+  "section_break_2",
+  "from_date",
```
Comment (Anurag810) [inline erpnext/erpnext_integrations/doctype/backfill_taxjar_transactions/backfill_taxjar_transactions.json:12]: why have you created a different section for only two fields you can put them together.

**Review by Anurag810 — DISMISSED** [review]
Please update description

File: erpnext/erpnext_integrations/taxjar_integration.py [hunk 6]
```diff
@@ -415,4 +458,4 @@ def get_line_item_dict(item, doc, refund_transaction=False):
 		product_tax_code = item.get("product_tax_code"),
 		sales_tax = item.get("tax_collectable"),
 		discount = line_item_discount + distributed_discount
-	)
\ No newline at end of file
+	)
```
Comment (sahil28297) [inline erpnext/erpnext_integrations/taxjar_integration.py]:
```suggestion
	)
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]
fix sider issues

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**